src/deep-search/lib/Client/Connection/Aws4SignedHttpConnectionFactory.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Created by CCristi <ccovali@mitocgroup.com> on 3/28/16.
+ */
+
+'use strict';
+
+import HttpConnection from 'elasticsearch/src/lib/connectors/http';
+import aws4 from 'aws4';
+
+/**
+ * Aws4 Signed Http Connection Factory
+ */
+export class Aws4SignedHttpConnectionFactory {
+  /**
+   * @param {Object} awsCredentials
+   */
+  constructor(awsCredentials = null) {
+    this._awsCredentials = awsCredentials;
+  }
+
+  /**
+   * @param {Object} aws4Signature
+   * @returns {$ES6_ANONYMOUS_CLASS$}
+   */
+  create() {
+    let connectionFactory = this;
+
+    /**
+     * Aws4 Signed Http Connection
+     */
+    return class extends HttpConnection {
+      /**
+       * @param {Object} params
+       * @returns {Object}
+       */
+      makeReqParams(params) {
+        let aws4Signature = connectionFactory._createAws4Signature({
+            ...params,
+            service: 'es'
+        });
+
+        params = super.makeReqParams(params);
+        params.headers = params.headers || {};
+
+        ['X-Amz-Date', 'X-Amz-Security-Token', 'Authorization'].forEach((header) => {
+          params.headers[header] = aws4Signature.headers[header];
+        });
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/MitocGroup/deep-framework/file/2320249722/issues/source?bid=2895475&fileBranchId=3204331#l47)